src/perform_step/extrapolation_perform_step.jl

```diff
@@ -283,7 +283,20 @@ function perform_step!(integrator,cache::ImplicitEulerExtrapolationCache,repeat_
         cache.linsolve[1](vec(k_tmps[1]), W[1], vec(linsolve_tmps[1]), !repeat_step)
         integrator.destats.nsolve += 1
         @.. k_tmps[1] = -k_tmps[1]
+        @.. u_tmps2[1] = u_tmps[1]
         @.. u_tmps[1] = u_tmps[1] + k_tmps[1]
+        if(index<=2 && j>=2)
```

Comment:
Just a minor comment regarding the style: it's usually more common (IIRC also throughout DiffEq) to not add brackets here, see, e.g., the [style guide in the docs](https://docs.julialang.org/en/v1/manual/style-guide/#Don't-parenthesize-conditions-1.) Hence I suggest to write instead
```suggestion
        if index<=2 && j>=2
```

Reply:
Thanks @devmotion for pointing in out, I didn't know about it. I will keep that in mind 😃 .